demos/projects/ESPRESSIF/adu/config/demo_config.h

```diff
@@ -218,10 +218,15 @@
  */
 #define democonfigIOTHUB_PORT            8883
 
+/**
+ * @brief Defines device restart function, used by the common sample modules.
+ */
+#define configRestartDevice()    esp_restart()
```

Comment:
Why do we name these "config"? /cc @danewalton @ewertons 
IMO, they better fit a platform abstraction rather than configuration.

Reply:
Agree - I was following the naming convention from the rest of the file, but I'm open to suggestions on the best name for this!

Reply:
I don't disagree. The naming convention is from FreeRTOS where you name #defines by prefixing with the filename. So in this case, `democonfigBLAH`.

We can surely create some kind of platform abstraction, but that would add a bunch more to this I think.